File: java/src/test/java/problems1_100/SearchA2dMatrixTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class SearchA2dMatrixTest extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "search-a-2d-matrix.json";
    }

    @Test
    public void test() {
        for (SearchA2dMatrix solution : SearchA2dMatrix.values()) {
            for (SearchA2dMatrixSample sample : getSamples(SearchA2dMatrixSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.searchMatrix(sample.input.matrix, sample.input.target))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/SearchA2dMatrixTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class SearchA2dMatrixTest : SampledTest() {
    override fun getSampleFilename(): String = "search-a-2d-matrix.json"

    @Test
    fun test() =
        SearchA2dMatrix.entries.forEach { solution ->
            getSamples(Array<SearchA2dMatrixSample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.searchMatrix(input.matrix, input.target))
                        .isEqualTo(output)
                }
        }
}
